networkx/algorithms/assortativity/connectivity.py

```diff
@@ -17,7 +17,7 @@ def _avg_deg_conn(G, neighbors, source_degree, target_degree,
     dsum = defaultdict(float)
     dnorm = defaultdict(float)
     for n,k in source_degree(nodes).items():
-        nbrdeg = target_degree(neighbors(n))
+        nbrdeg = target_degree(list(neighbors(n)))
```

Comment:
OK I think I got it, 
``` python
In [1]: import networkx as nx
In [2]: G=nx.path_graph(4)
In [3]: D = G.to_directed()
In [5]: D.degree(D.neighbors(0))
Out[5]: {}
In [6]: D.degree(list(D.neighbors(0)))
Out[6]: {1: 4}
In [9]: D.degree(list(D.neighbors(1)))
Out[9]: {0: 2, 2: 4}
In [10]: D.degree((D.neighbors(1)))
Out[10]: {0: 3}
In [25]: D.neighbors(0)
Out[25]: <dictionary-keyiterator at 0x111282890>
In [30]: D.degree(D.edges_iter(0))
Out[30]: {}
In [31]: D.degree(list(D.edges_iter(0)))
Out[31]: {}
In [28]: D.edges_iter(0)
Out[28]: <generator object edges_iter at 0x111259b90>
```
I think this is because neighbors returns a dictionary-keyiterator. What do you think @dschult ?